apps/fxc-front/src/app/logic/score/scorer.ts

```diff
@@ -1,19 +1,104 @@
-import { CircuitType } from '@flyxc/optimizer/lib/api';
-
-export class Score {
-  distanceM: number;
-  indexes: number[];
-  multiplier: number;
-  circuit: CircuitType;
-  closingRadiusKm: number;
-  points: number;
-
-  constructor(score: Partial<Score>) {
-    this.distanceM = score.distanceM ?? 0;
-    this.indexes = score.indexes ?? [];
-    this.multiplier = score.multiplier ?? 1;
-    this.circuit = score.circuit ?? CircuitType.OpenDistance;
-    this.closingRadiusKm = score.closingRadiusKm ?? 0;
-    this.points = score.points ?? 0;
+import type { LatLonAltTime, ScoringResult } from '@flyxc/optimizer/lib/optimizer';
+
+import { type Request as WorkerRequest, type Response as WorkerResponse } from '../../workers/optimizer';
+import ScoringWorker from '../../workers/optimizer?worker';
+import { getScoringRuleName, type LeagueCode } from './league/leagues';
+
+export enum ScoreOrigin {
+  INTERACTIVE = 'interactive',
+  TRACK = 'track',
+}
+
+export type Score = ScoringResult & { origin: ScoreOrigin };
+
+export type ScoringResultHandler = (result: ScoringResult) => void;
+
+export type ScoringRequestIdProvider = () => number;
+
+export class Scorer {
+  private scoringWorker: Worker | null = null;
+  private readonly handleScoringResult: ScoringResultHandler;
+  private readonly getScoringRequestId: ScoringRequestIdProvider;
+
+  /**
+   * Builds a new Scorer with the two required functions.
```

Comment:
nit: 
- I don't think "build" is the correct semantic here, maybe instantiate?
- Also we know that a ctor instantiate
- And we know from the signature that it takes 2 functions
- Repeating trivial info is useless as it doesn't add value and might even get out of sync if the signature is updated

-> Frop this line entirely?